src/NUnitFramework/framework/Attributes/CombiningStrategyAttribute.cs

```diff
@@ -75,10 +75,12 @@ protected CombiningStrategyAttribute(object strategy)
         /// Construct one or more TestMethods from a given MethodInfo,
         /// using available parameter data.
         /// </summary>
+        /// <param name="fixtureType">The parameter containing type of the test fixture class. 
+        /// This may be different from the reflected member info</param>
         /// <param name="method">The MethodInfo for which tests are to be constructed.</param>
         /// <param name="suite">The suite to which the tests will be added.</param>
         /// <returns>One or more TestMethods</returns>
-        public IEnumerable<TestMethod> BuildFrom(MethodInfo method, Test suite)
+        public IEnumerable<TestMethod> BuildFrom(Type fixtureType, MethodInfo method, Test suite)
```

Comment:
In general, you probably don't need both a type argument and a fixture argument to any method. There is some indication in the code that suite may be null, but I think that may no longer be true. If we use the suite, we could introduce a guard clause and see if anything fails.